index.js

```diff
@@ -126,7 +126,9 @@ function replacePath (path, keys) {
       name: key || index++,
       delimiter: prefix || '/',
       optional: optional,
-      repeat: repeat
+      repeat: repeat,
+      offset: i + (prefix ? prefix.length : 0),
```

Comment:
I think it should be, otherwise you need to minus 1 every time. This also allows you to get length with just `_.length` (rename that to `all` or something useful though). If you want, go ahead. I was starting to write it too and it's pretty neat.